src/commands/pczt/create_manual.rs

```diff
@@ -0,0 +1,414 @@
+        // TODO: Handle change outputs. For now, you are required to spend it all.
+        let value_out = (value_in - fee).ok_or_else(|| anyhow!("Balance underflow"))?;
+        if let Some(v) = value {
+            if v != value_out {
+                return Err(anyhow!(
+                    "Change is not currently supported. Don't set --value, or set it to '{}'",
+                    value_out.into_u64(),
+                ));
+            }
+        }
+
+        // Now construct the real builder.
+        let builder = prepare_builder(transparent_inputs, recipient.clone(), value_out, memo)?;
+
+        let PcztResult { pczt_parts, .. } =
+            builder.build_for_pczt(rng, &zip317::FeeRule::standard())?;
+        let created = Creator::build_from_parts(pczt_parts)
+            .ok_or_else(|| anyhow!("Transaction version is incompatible with PCZTs"))?;
+
+        let io_finalized = IoFinalizer::new(created)
+            .finalize_io()
+            .map_err(|e| anyhow!("{e:?}"))?;
+
+        let pczt = handle_recipient(
+            recipient,
+            (Updater::new(io_finalized), self.address),
+            |_, (updater, user_address)| {
+                updater
+                    .update_transparent_with(|mut u| {
+                        assert_eq!(u.bundle().outputs().len(), 1);
+                        u.update_output_with(0, |mut ou| {
+                            ou.set_user_address(user_address);
+                            Ok(())
+                        })
+                    })
+                    .map_err(|e| anyhow!("{e:?}"))
+            },
+            |_, (updater, user_address)| {
+                updater
+                    .update_sapling_with(|mut u| {
+                        assert_eq!(u.bundle().outputs().len(), 1);
+                        u.update_output_with(0, |mut ou| {
+                            ou.set_user_address(user_address);
+                            Ok(())
+                        })
+                    })
+                    .map_err(|e| anyhow!("{e:?}"))
+            },
+            |_, (updater, user_address)| {
+                updater
+                    .update_orchard_with(|mut u| {
+                        // Because of padding, we need to find the action that contains
+                        // the output. We could do this with the Orchard bundle metadata,
+                        // but as there is only one real output we can just look for it.
+                        assert_eq!(u.bundle().actions().len(), 2);
+                        let index =
+                            u.bundle()
+                                .actions()
+                                .iter()
+                                .enumerate()
+                                .find_map(|(i, action)| {
+                                    action.output().value().and_then(|v| {
+                                        (v.inner() == value_out.into_u64()).then_some(i)
+                                    })
```

Comment:
This only looks for the output by value, I think. In which case, if there were two outputs with the same value, it might find the wrong one.